pyof/foundation/basic_types.py

```diff
@@ -478,33 +478,22 @@ class attribute of the struct.
         return len(value)
 
 
-class TypeList(list, GenericStruct):
-    """Base class for lists that store objects of one single type."""
-
-    def __init__(self, items):
-        """Initialize the list with one item or a list of items.
+class FixedTypeList(list, GenericStruct):
+    """A list that stores instances of one pyof class."""
 
-        Args:
-            items (iterable, ``pyof_class``): Items to be stored.
-        """
-        super().__init__()
-        if isinstance(items, list):
-            self.extend(items)
-        elif items:
-            self.append(items)
+    _pyof_class = None
 
-    def extend(self, items):
-        """Extend the list by adding all items of ``items``.
+    def __init__(self, pyof_class, items=None):
+        """The constructor parameters follows.
 
         Args:
-            items (iterable): Items to be added to the list.
-
-        Raises:
-            :exc:`~.exceptions.WrongListItemType`: If an item has an unexpected
-                type.
+            pyof_class (:obj:`type`): Class of the items to be stored.
+            items (iterable, ``pyof_class``): Items to be stored.
         """
-        for item in items:
-            self.append(item)
+        self._pyof_class = pyof_class
+        if items is None:
+            items = []
+        super().__init__(items)
```

Comment:
1: This is the same behavior of any of the other classes. UBInt8('a') for example is perfectly valid, and it only complains when trying to pack. The list is just keeping the same behavior.

2: This allows FixedTypeList(UBInt8, [1,2,3]) for example. Checking the class does not.